esrally/driver/runner.py

```diff
@@ -1320,46 +1329,56 @@ async def _wait_for(self, es, idx, description):
             node_info = await es.nodes.info()
             for node in node_info["nodes"].values():
                 if "data" in node["roles"]:
                     node_names.append(node["name"])
             if not node_names:
                 raise exceptions.RallyAssertionError("Could not choose a suitable shrink-node automatically. Please specify it explicitly.")
+
+        for source_index in source_indices:
             shrink_node = random.choice(node_names)
-        self.logger.info("Using [%s] as shrink node.", shrink_node)
-        self.logger.info("Preparing [%s] for shrinking.", source_index)
-        # prepare index for shrinking
-        await es.indices.put_settings(index=source_index,
-                                      body={
-                                          "settings": {
-                                              "index.routing.allocation.require._name": shrink_node,
-                                              "index.blocks.write": "true"
-                                          }
-                                      },
-                                      preserve_existing=True)
-
-        self.logger.info("Waiting for relocation to finish for index [%s]...", source_index)
-        await self._wait_for(es, source_index, "shard relocation for index [{}]".format(source_index))
-        self.logger.info("Shrinking [%s] to [%s].", source_index, target_index)
-        if "settings" not in target_body:
-            target_body["settings"] = {}
-        target_body["settings"]["index.routing.allocation.require._name"] = None
-        target_body["settings"]["index.blocks.write"] = None
-        # kick off the shrink operation
-        await es.indices.shrink(index=source_index, target=target_index, body=target_body)
-
-        self.logger.info("Waiting for shrink to finish for index [%s]...", source_index)
-        await self._wait_for(es, target_index, "shrink for index [{}]".format(target_index))
-        self.logger.info("Shrinking [%s] to [%s] has finished.", source_index, target_index)
+            self.logger.info("Using [%s] as shrink node.", shrink_node)
+            self.logger.info("Preparing [%s] for shrinking.", source_index)
+
+            # prepare index for shrinking
+            await es.indices.put_settings(index=source_index,
+                                          body={
+                                              "settings": {
+                                                  "index.routing.allocation.require._name": shrink_node,
+                                                  "index.blocks.write": "true"
+                                              }
+                                          },
+                                          preserve_existing=True)
+
+            self.logger.info("Waiting for relocation to finish for index [%s] ...", source_index)
+            await self._wait_for(es, source_index, "shard relocation for index [{}]".format(source_index))
```

Comment:
We could use an f-string here?

Reply:
Done in 52f28a6e3729b9a6f9006bcb56099a5e0efc9836